packages/@aws-cdk/aws-appmesh/lib/gateway-route.ts

```diff
@@ -0,0 +1,390 @@
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnGatewayRoute } from './appmesh.generated';
+import { Protocol } from './shared-interfaces';
+import { IVirtualGateway, VirtualGateway } from './virtual-gateway';
+import { IVirtualService } from './virtual-service';
+
+/**
+ * Interface for which all GatewayRoute based classes MUST implement
+ */
+export interface IGatewayRoute extends cdk.IResource {
+  /**
+   * The name of the GatewayRoute
+   *
+   * @attribute
+   */
+  readonly gatewayRouteName: string,
+
+  /**
+   * The Amazon Resource Name (ARN) for the GatewayRoute
+   *
+   * @attribute
+   */
+  readonly gatewayRouteArn: string;
+
+  /**
+   * The VirtualGateway the GatewayRoute belongs to
+   *
+   * @attribute
+   */
+  readonly virtualGateway: IVirtualGateway;
+}
+
+/**
+ * Basic configuration properties for a GatewayRoute
+ */
+export interface GatewayRouteBaseProps {
+  /**
+   * The name of the GatewayRoute
+   *
+   * @default - an automatically generated name
+   */
+  readonly gatewayRouteName?: string;
+
+  /**
+   * What protocol the route uses
+   */
+  readonly routeSpec: GatewayRouteSpec;
+}
+
+/**
+ * The criterion for determining a request match for this GatewayRoute
+ */
+export interface HttpGatewayRouteMatch {
+  /**
+   * Specifies the path to match requests with.
+   * This parameter must always start with /, which by itself matches all requests to the virtual service name.
+   * You can also match for path-based routing of requests. For example, if your virtual service name is my-service.local
+   * and you want the route to match requests to my-service.local/metrics, your prefix should be /metrics.
+   *
+   */
+  readonly prefixPath: string;
+}
+
+/**
+ * The criterion for determining a request match for this GatewayRoute
+ */
+export interface GrpcGatewayRouteMatch {
+  /**
+   * The fully qualified domain name for the service to match from the request
+   */
+  readonly serviceName: string;
+}
+
+/**
+ * Properties to define a new GatewayRoute
+ */
+export interface GatewayRouteProps extends GatewayRouteBaseProps {
+  /**
+   * The VirtualGateway this GatewayRoute is associated with
+   */
+  readonly virtualGateway: IVirtualGateway;
+}
+
+/**
+ * Properties specific for HTTP Based GatewayRoutes
+ */
+export interface HttpRouteSpecProps {
+  /**
+   * The criterion for determining a request match for this GatewayRoute
+   *
+   * @default - matches on '/'
+   */
+  readonly match?: HttpGatewayRouteMatch;
+  /**
+   * The VirtualService this GatewayRoute directs traffic to
+   */
+  readonly routeTarget: IVirtualService;
```

Comment:
Please, make sure there's an empty line between properties (you have it in many of them, but not all).

```suggestion
  /**
   * The criterion for determining a request match for this GatewayRoute
   *
   * @default - matches on '/'
   */
  readonly match?: HttpGatewayRouteMatch;
  
  /**
   * The VirtualService this GatewayRoute directs traffic to
   */
  readonly routeTarget: IVirtualService;
```

Reply:
done!